crate/cli/src/actions/findex/mod.rs

```diff
@@ -1,3 +1,4 @@
 pub mod insert_or_delete;
 pub mod parameters;
 pub mod search;
+pub(crate) const MAX_SEMAPHORES: usize = 1024; // base value for most linux kernels
```

Comment:
This comments is not clear enough: which base value are you referring to? The base semaphore value in the linux kernel?
```suggestion
/// This constant defines the number of concurrent calls to the server that can be performed
/// per CLI-command invocation.
///
/// Each call to the server opens a new file descriptor. Since most Linux distributions allow for up
/// to 1024 concurrently-open file descriptors, limiting the number of concurrent call is necessary 
/// to avoid reaching system limits.
pub(crate) const MAX_SEMAPHORES: usize = 1024;
```

Btw, 1024 may be slightly too big : user sessions usually have a few open file descriptors. An acceptable limit may therefore be less than 1024. I suggest some tests are performed with the file-descriptor limit properly set.

Reply:
@Manuthor / @bgrieder  what do you think ? I think the topic have been discussed yesterday

On my end I did some tests : starting from ~100 semaphores **there is no performance gain** the exact number is a bit below 100 - but any bigger values do do not gain anything ( To make this conclusion, I restarted the test multiple times with diff values on an i9 machine, Ubuntu 22.04 LTS equivalent )

Reply:
I also changed the suggested description  : 

 `/// Maximum number of concurrent network calls allowed per CLI findex-command invocation.
///
/// Each network call opens a socket which consumes a file descriptor. While the system's file descriptor
/// limit can be configured (via `ulimit -n` or `/etc/security/limits.conf`), we enforce this fixed limit
/// which is the default soft limit on many Linux systems to ensure consistent behavior.`

Reply:
> to ensure consistent behavior

this is not the point of this constant : what we aimed for was to avoid the error thrown by the OS upon reaching the limit of open FDs.